specs/data_structures.md

```diff
@@ -413,11 +419,11 @@ As an example, the parity data in the second column of `Q2` (in striped purple)
 
 ![fig: RS2D encoding: extending a column.](./figures/rs2d_extend.svg)
 
-Now that all four quadrants of the `2k * 2k` matrix are filled, the row and column roots can be computed. To do so, each row/column is used as the leaves of a [NMT](#namespace-merkle-tree), for which the compact root is computed (i.e. an extra hash operation is used to produce a single [HashDigest](#hashdigest)). In this example, the fourth row root value is computed as the NMT root of the fourth row of `Q0` and the fourth row of `Q1` as leaves.
+Now that all four quadrants of the `2k * 2k` matrix are filled, the row and column roots can be computed. To do so, each row/column is used as the leaves of a [NMT](#namespace-merkle-tree), for which the compact root is computed (i.e. an extra hash operation over the NMT root is used to produce a single [HashDigest](#hashdigest)). In this example, the fourth row root value is computed as the NMT root of the fourth row of `Q0` and the fourth row of `Q1` as leaves.
```

Comment:
Question: Is the compact root a hard requirement here? Especially given that a single and compact commitment to all roots will end up as `availableDataRoot` in the header.

Reply:
It's not a hard requirement, for data availability proofs, however it does reduce the size of the commitments that need to be downloaded (which over time adds up). Nodes have to download some commitment to each NMT they sample into, so might as well make that as small as possible.

Reply:
What is a compact root? This is the only time a reference to compact root is mentioned. How does an extra hash operation result in a compact root?

Reply:
The compact root is, as described, the hash of the NMT root (which is a tuple (min namespace ID, max namespace ID, hash). I should probably use the expression "compact root" explicitly, but they're used for the row and column roots in the available data header.

Reply:
Fixed in ccffc28.